### PR TITLE
Feature - api model schema

### PR DESCRIPTION
**Pull request by at-huytong**

### CODE REVIEW

File: MyApp/Model/Schema/City.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//  Copyright © 2018 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+import ObjectMapper
+import RealmS
+
+@objcMembers final class City: Object, StaticMappable {
+
```
Comment (at-daiho) [inline MyApp/Model/Schema/City.swift]: WARNING - Lines should not have trailing whitespace.

File: MyApp/Model/Schema/City.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//
+
+import Foundation
+import RealmSwift
+import ObjectMapper
+import RealmS
+
+@objcMembers final class City: Object, StaticMappable {
+    
+    dynamic var id = 0
```
Comment (at-daiho) [inline MyApp/Model/Schema/City.swift]: WARNING - Variable name should be between 3 and 40 characters long: 'id'

File: MyApp/Model/Schema/Forecast.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+//
+
+import Foundation
+import RealmSwift
+import ObjectMapper
+import RealmS
+
+@objcMembers final class Forecast: Object, Mappable {
+
+    dynamic var dt: Double = 0
```
Comment (at-daiho) [inline MyApp/Model/Schema/Forecast.swift]: WARNING - Variable name should be between 3 and 40 characters long: 'dt'

File: MyApp/Model/Schema/ForecastResponse.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+//
+
+import Foundation
+import RealmSwift
+import ObjectMapper
+import RealmS
+
+@objcMembers final class ForecastResponse: Object, StaticMappable {
+
+    dynamic var id = 0
```
Comment (at-daiho) [inline MyApp/Model/Schema/ForecastResponse.swift]: WARNING - Variable name should be between 3 and 40 characters long: 'id'

File: MyApp/Model/Schema/Weather.swift [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+//
+
+import Foundation
+import RealmSwift
+import ObjectMapper
+import RealmS
+
+@objcMembers final class Weather: Object, StaticMappable {
+
+    dynamic var id = 0
```
Comment (at-daiho) [inline MyApp/Model/Schema/Weather.swift]: WARNING - Variable name should be between 3 and 40 characters long: 'id'

**Comment by at-daiho** [comment]
Total linter violations in pull request: 5
Serious: 0
Warnings: 5

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Model/API/Core/Api.swift [hunk 6]
```diff
@@ -10,7 +10,22 @@ import Foundation
...
+
+    struct Forecast {
+    }
+}
+
+extension Api.Path {
+    struct Forecast: ApiPath {
+        static var path: String { return Api.Path.baseURL / "forecast" / "daily" }
+        var city: String
+        var urlString: String { return Forecast.path / "?q=\(city)&units=metric&appid=\(Config.openWeatherMapAppId)" }
```
Comment (at-quangphu) [inline MyApp/Model/API/Core/Api.swift]: `q=\(city)&units=metric&appid=\(Config.openWeatherMapAppId)` cái này là parameters của url. Chỗ này mình chỉ define `path` thôi nha.

File: MyApp/Model/Schema/City.swift [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    override static func primaryKey() -> String? {
+        return "cityId"
+    }
+
+    func mapping(map: Map) {
+        name <- map["name"]
+        country <- map["country"]
+    }
+
+    static func objectForMapping(map: Map) -> BaseMappable? {
```
Comment (at-quangphu) [inline MyApp/Model/Schema/City.swift]: Func này để làm gì vậy Huy?

File: MyApp/Model/API/Core/Api.swift [hunk 8]
```diff
@@ -10,7 +10,21 @@ import Foundation
...
+    struct Weather {
+    }
+
+    struct Forecast {
+    }
+}
+
+extension Api.Path {
+    struct Forecast: ApiPath {
+        static var path: String { return Api.Path.baseURL / "forecast" / "daily" }
```
Comment (at-quangphu) [inline MyApp/Model/API/Core/Api.swift]: Nên tạo ra 1 biến store path cho `forecast daily` nha.
```swift
static var path: String { return Api.Path.baseURL / "forecast" }
static var daily: String { return Forecast.path / "daily" }
```

File: MyApp/Model/API/Core/Api.swift [hunk 9]
```diff
@@ -23,8 +23,9 @@ final class Api {
 
 extension Api.Path {
     struct Forecast: ApiPath {
-        static var path: String { return Api.Path.baseURL / "forecast" / "daily" }
-        var urlString: String { return Forecast.path }
+        static var path: String { return Api.Path.baseURL / "forecast" }
+        static var daily: String { return Forecast.path / "daily" }
+        var urlString: String { return Forecast.daily }
```
Comment (at-quangphu) [inline MyApp/Model/API/Core/Api.swift]:
```swift
var urlString: String { return Forecast.path }
```

**Pull request merged**